packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -113,25 +123,41 @@ export class ActionDestination implements DestinationPlugin {
     return this.action.isLoaded()
   }
 
-  ready(): Promise<unknown> {
-    return this.action.ready ? this.action.ready() : Promise.resolve()
+  async ready(): Promise<boolean> {
+    const loadOutput = await this.loadPromise.promise
+    return !(loadOutput instanceof Error)
   }
 
   async load(ctx: Context, analytics: Analytics): Promise<unknown> {
+    if (this.loadPromise.isSettled()) {
+      const prevLoad = await this.loadPromise.promise
+      if (prevLoad instanceof Error) {
+        throw prevLoad
+      } else {
+        return prevLoad
+      }
+    }
+
     try {
       recordIntegrationMetric(ctx, {
         integrationName: this.action.name,
         methodName: 'load',
         type: 'action',
       })
-      return await this.action.load(ctx, analytics)
+
+      const loadP = this.action.load(ctx, analytics)
+
+      this.loadPromise.resolve(await loadP)
+      return loadP
     } catch (error) {
       recordIntegrationMetric(ctx, {
         integrationName: this.action.name,
         methodName: 'load',
         type: 'action',
         didError: true,
       })
+
+      this.loadPromise.resolve(error)
```

Comment:
Nit: Why not reject? Looks a bit strange!

Then wouldn't need the error instanceof Error thing

Reply:
Very good catch mate. Thank you!

This was by design back in the day when I didn't know about the trick of using `Promise.allSettled([promise])` based "proxying" in `createDeffered` utility. Using `promise.finally` directly there, and then rejecting the same promise somewhere else would cause "Unhandled promise rejection" even if you added `catch` trap. That was because `catch` would've been added after `finally` hook was.

Now, with the new technique this behaviour is not exhibitted since we "fork" the promise instead of attachting traps directly.

Code has been updated and loadPromise now rejects. Thanks again for highlighting this.